src/exchange/contractService.ts

```diff
@@ -518,4 +569,104 @@ export class ContractCalls {
   getSUIBalance = async (walletAddress?: string): Promise<string> => {
     return await this.onChainCalls.getUserSuiBalance(walletAddress);
   };
+
+  closeAllPositionsAndWithdrawMarginPTB = async (
+    delistedMarketPositions: string[],
+    args?: {
+      gasBudget?: number;
+      sponsor?: boolean;
+      dryRunOnly?: boolean;
+    }
+  ) => {
+    return TransformToResponseSchema(
+      async () => {
+        try {
+          return await this.onChainCalls.closeAllPositionsAndWithdrawMarginPTB(
+            delistedMarketPositions,
+            this.walletAddress || this.signer.toSuiAddress(),
+            args
+          );
+        } catch (error) {
+          return error;
+        }
+      },
+      interpolate(SuccessMessages.closedDelistedPositionsAndWithdrawMargin, {
+        amount: "all",
+      }),
+      args?.sponsor
+    );
+  };
+  closeAllPositionsdWithdrawSwapAndDepositToProPTB = async (
+    delistedMarketPositions: string[],
+    args?: {
+      gasBudget?: number;
+      sponsor?: boolean;
+      use7k?: boolean;
+      slippage?: number;
+      dryRunOnly?: boolean;
+      defaultSlippage?: number;
+    }
+  ) => {
+    return TransformToResponseSchema(async () => {
+      try {
+        return await this.onChainCalls.closeAllPositionsWithdrawSwapAndDepositToProPTB(
+          delistedMarketPositions,
+          this.spotOnchain,
+          this.proOnchain,
+          this.walletAddress || this.signer.toSuiAddress(),
+          args
+        );
+      } catch (error) {
+        return error;
+      }
+    }, interpolate(SuccessMessages.closedDelistedPositionsSwapAndDepositToPro, { amount: "all" }));
+  };
+
+  withdrawAllSwapAndDepositToProPTB = async (args?: {
```

Comment:
Why do we have this and `withdrawAllWusdcSwapAndDepositToPro`?

Reply:
Created it for testing without closing delisted position on prod to verify rest of the calls work fine when combined together